### PR TITLE
Refactor module ownership or access check using a new selector

### PR DESCRIPTION
**Pull request by kuasha420**
## Summary

Addresses issue:

- #6192 

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by hussain-t — APPROVED** [review]
Nice one, @kuasha420. LGTM 👍

**Review by hussain-t — CHANGES_REQUESTED** [review]
@kuasha420, though no changes affect the UI or features, it’s better to test the areas that were refactored and do a smoke test. For example, the modules’ settings header and the settings edit page.  Could you add a QAB?

**Comment by kuasha420** [comment]
@hussain-t Thanks for spotting the missing QA brief. Added and moved it to MR. :+1:

**Review by hussain-t — APPROVED** [review]
Approved after verifying the QAB.

**Review by tofumatt — CHANGES_REQUESTED** [review]
Looks great, just a few comments and then this is good to go 😄

File: assets/js/googlesitekit/modules/datastore/modules.js [hunk 1]
```diff
@@ -1238,6 +1238,42 @@ const baseSelectors = {
...
+	/**
+	 * Checks if current user has ownership or access to the given module.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @param {string} slug  Module slug.
+	 * @return {(boolean|undefined)} `true` if the user has ownership or access.
+	 * 								 `false` if the user doesn't have ownership or access.
+	 * 								 `undefined` If the state is still being resolved,
```
[the remark this answers is not in the document]
Comment (kuasha420) [reply assets/js/googlesitekit/modules/datastore/modules.js:1250]: Yikes, thanks.

File: assets/js/googlesitekit/modules/datastore/modules.js [hunk 2]
```diff
@@ -1238,6 +1238,42 @@ const baseSelectors = {
...
+	hasModuleOwnershipOrAccess: createRegistrySelector(
+		( select ) => ( state, moduleSlug ) => {
+			const moduleStoreName =
+				select( CORE_MODULES ).getModuleStoreName( moduleSlug );
+
+			if ( moduleStoreName === undefined ) {
+				return undefined;
+			}
+
+			const moduleOwnerID = select( moduleStoreName )?.getOwnerID();
```
[the remark this answers is not in the document]
Comment (kuasha420) [reply assets/js/googlesitekit/modules/datastore/modules.js:1261]: Sounds Reasonable, applied!

[the remark this answers is not in the document]
Comment (kuasha420) [reply assets/js/googlesitekit/modules/datastore/modules.js]: Hmm, turns out `select` call for a non-existent module returns `null` and not `undefined`. I have changed the check to a `null`. Let me know what you think.

**Comment by kuasha420** [comment]
@tofumatt Thank you for the review, addressed your feedback and fixed up the merge conflicts/test fails. Back to you for another look. Cheers.

File: assets/js/googlesitekit/modules/datastore/modules.js [hunk 3]
```diff
@@ -1239,6 +1239,49 @@ const baseSelectors = {
...
+			if ( moduleStoreName === undefined ) {
+				return undefined;
+			}
+
+			// A store with this name doesn't exist, so the user can't have access to it.
+			// This is either caused by a module not being loaded or an incorrect module
+			// name being used.
+			if ( select( moduleStoreName ) === null ) {
+				return false;
+			}
```
Comment (tofumatt) [inline assets/js/googlesitekit/modules/datastore/modules.js:1267]: There wasn't a test for this behaviour, but it was easy to add one so I did: 32df08712130c346b51203e1064710f73573d5a4

**Review by tofumatt — APPROVED** [review]

**Pull request merged**